### PR TITLE
New module openms/idfilter

### LINKED ISSUE #3993: new module: openms/idfilter
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jonasscheid**
## PR checklist

Closes #3993 

Implements another OpenMS module [IDFilter](https://abibuilder.cs.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_IDFilter.html). The module has an optional filter file as input as well as variable filter criteria (see docs). I needed to set a default filter criterion for the optional filter file.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/openms/idfilter/main.nf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_2' :
+        'biocontainers/openms:3.1.0--h8964181_2' }"
+
+    input:
+    tuple val(meta), path(id_file), path(filter_file)
+
+    output:
+    tuple val(meta), path("*.{idXML,consensusXML}"), emit: filtered
+    path "versions.yml"                             , emit: versions
```
Comment (FriederikeHanssen) [inline modules/nf-core/openms/idfilter/main.nf:15]:
```suggestion
    path "versions.yml"                            , emit: versions
```

File: modules/nf-core/openms/idfilter/main.nf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def suffix = task.ext.suffix ?: "${id_file.getExtension()}"
+    // Optional filtering via filter_file
+    def filter_citerion = task.ext.args2 ?: "-whitelist:peptides"
```
Comment (FriederikeHanssen) [inline modules/nf-core/openms/idfilter/main.nf:25]: can you combine this with the next line? what do you need the second args for?

File: modules/nf-core/openms/idfilter/main.nf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    def filter = filter_file ? "${filter_citerion} ${filter_file}" : ""
```
Comment (FriederikeHanssen) [inline modules/nf-core/openms/idfilter/main.nf:26]:
```suggestion
    def filter = filter_file ? "-whitelist:peptides ${filter_file}" : ""
```

Comment (jonasscheid) [reply modules/nf-core/openms/idfilter/main.nf:26]: There can be multiple filter criteria, e.g. `-whitelist:peptides|proteins`, `-blacklist:peptides|proteins` `-in_silico_digestion:fasta`. I guess this needs to be defined in ` task.ext.args2` 🤔 or how would you do it?

Comment (jonasscheid) [reply modules/nf-core/openms/idfilter/main.nf:26]: See [doc](https://abibuilder.cs.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_IDFilter.html)

Comment (FriederikeHanssen) [reply modules/nf-core/openms/idfilter/main.nf:26]: ah ok then it makes sense :)

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**